### PR TITLE
Add suitest for controller

### PR DESCRIPTION
**Pull request by Aisuko**
**Description**

This PR relates to #463 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Aisuko** [comment]
We need to add further steps in CI for setting up the test environment. Otherwise the Code coverage would be failed.

**Comment by Aisuko** [comment]
```
@Aisuko ➜ /workspaces/meshery-operator (feat/reconcile_coverage) $ make coverage
go test -v ./... -coverprofile cover.out
?       github.com/layer5io/meshery-operator    [no test files]
=== RUN   TestAPIs
Running Suite: APIs Suite - /workspaces/meshery-operator/api/v1alpha1
=====================================================================
Random Seed: 1684120687

Will run 11 of 11 specs
•••••••••••

Ran 11 of 11 Specs in 0.008 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (0.01s)
PASS
        github.com/layer5io/meshery-operator/api/v1alpha1       coverage: 43.8% of statements
ok      github.com/layer5io/meshery-operator/api/v1alpha1       0.070s  coverage: 43.8% of statements
?       github.com/layer5io/meshery-operator/pkg/broker [no test files]
?       github.com/layer5io/meshery-operator/pkg/client [no test files]
?       github.com/layer5io/meshery-operator/pkg/client/v1alpha1        [no test files]
?       github.com/layer5io/meshery-operator/pkg/meshsync       [no test files]
=== RUN   TestAPIs
Running Suite: Controller Suite - /workspaces/meshery-operator/controllers
==========================================================================
Random Seed: 1684120692

Will run 6 of 6 specs
••••••W0515 03:18:19.032841   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:18:19.032895   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
W0515 03:18:19.914974   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:18:19.915056   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
W0515 03:18:22.562162   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:18:22.562229   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
W0515 03:18:27.569295   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:18:27.569346   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
W0515 03:18:40.188180   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:18:40.188217   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
W0515 03:19:02.382254   27109 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused
E0515 03:19:02.382290   27109 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.3/tools/cache/reflector.go:169: Failed to watch *v1.StatefulSet: failed to list *v1.StatefulSet: Get "https://127.0.0.1:37107/apis/apps/v1/statefulsets?limit=500&resourceVersion=0": dial tcp 127.0.0.1:37107: connect: connection refused


Ran 6 of 6 Specs in 66.956 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (66.96s)
PASS
        github.com/layer5io/meshery-operator/controllers        coverage: 22.8% of statements
ok      github.com/layer5io/meshery-operator/controllers        67.037s coverage: 22.8% of statements
go tool cover -html=cover.out -o cover.html
```

**Comment by leecalcote** [comment]
@Azanul

**Comment by Aisuko** [comment]
HI, guys. Shall we merge this one?

File: api/v1alpha1/meshsync_types_test.go [hunk 1]
```diff
@@ -94,28 +94,6 @@ var _ = Describe("The test case for the meshsync CRDs", func() {
 			Expect(meshSyncGet.Spec.Size == 5).Should(BeTrue())
 		})
 
-		It("The meshsync CRDs update the status of the resources", func() {
```
Comment (Azanul) [inline api/v1alpha1/meshsync_types_test.go:97]: Any reason for this to be removed?

Comment (Aisuko) [reply api/v1alpha1/meshsync_types_test.go]: Nice catch, I check this one and add it back.

Comment (Aisuko) [reply api/v1alpha1/meshsync_types_test.go]: @Azanul Can we merge this now?

**Comment by Azanul** [comment]
LGTM

**Review by Azanul — APPROVED** [review]

**Comment by Aisuko** [comment]
> LGTM

Thank you

**Pull request merged**